planner/core/preprocess.go

```diff
@@ -833,12 +833,34 @@ func (p *preprocessor) checkDropSequenceGrammar(stmt *ast.DropSequenceStmt) {
 }
 
 func (p *preprocessor) checkDropTableGrammar(stmt *ast.DropTableStmt) {
-	p.checkDropTableNames(stmt.Tables)
 	enableNoopFuncs := p.ctx.GetSessionVars().EnableNoopFuncs
 	if stmt.TemporaryKeyword == ast.TemporaryLocal && !enableNoopFuncs {
 		p.err = expression.ErrFunctionsNoopImpl.GenWithStackByArgs("DROP TEMPORARY TABLE")
 		return
 	}
+	currentDB := model.NewCIStr(p.ctx.GetSessionVars().CurrentDB)
+	if stmt.TemporaryKeyword == ast.TemporaryNone {
+		return
+	}
+	for _, t := range stmt.Tables {
+		if isIncorrectName(t.Name.String()) {
+			p.err = ddl.ErrWrongTableName.GenWithStackByArgs(t.Name.String())
+			return
+		}
+		if t.Schema.String() != "" {
+			currentDB = t.Schema
+		}
+		tableInfo, err := p.ensureInfoSchema().TableByName(currentDB, t.Name)
+		if err != nil {
+			p.err = err
+			return
+		}
+		if tableInfo.Meta().TempTableType == model.TempTableNone {
+			p.err = ErrOptOnTemporaryTable.GenWithStackByArgs("drop temporary table")
```

Comment:
What will the error message be? `drop temporary table is unsupported on temporary tables`?

Reply:
yes, do we need to change the words?

Reply:
We need another error message. This error message is not suitable for this case.